File: sliding-window/max_sum_of_contingous_subarray_of_size_n/python/MainApp/app/mainApp.py
import sys


class MainApp:
    def __init__(self):
        pass

    def run(self, arr, k):
        max_value = -sys.maxsize - 1
        current_running_sum = 0

        for idx, ele in enumerate(arr):
            current_running_sum += arr[idx]

            if idx >= k - 1:
                max_value = max(current_running_sum, max_value)
                current_running_sum -= arr[idx - (k - 1)]
        return max_value

File: sliding-window/max_sum_of_contingous_subarray_of_size_n/java/MainApp/src/main/java/io/github/mainapp/MainApp.java
package io.github.mainapp;

public class MainApp {
    public int Run(int [] arr, int k) {
        int maxValue = Integer.MIN_VALUE;
        int currentRunningSum = 0;

        for (int i = 0; i < arr.length; i++)
        {
            currentRunningSum += arr[i];
            if (i >= k - 1)
            {
                maxValue = Math.max(maxValue, currentRunningSum);
                currentRunningSum -= arr[i - (k - 1)];
            }
        }
        return maxValue;
    }
}
